src/main/java/agh/ics/oop/RectangularMap.java

```diff
@@ -0,0 +1,58 @@
+package agh.ics.oop;
+
+import java.util.ArrayList;
+
+public class RectangularMap implements IWorldMap {
+    public final int WIDTH;
+    public final int HEIGHT;
+    public final Vector2d ORIGIN;
+    public final Vector2d BOUNDARY;
+    private final ArrayList<Animal> animals = new ArrayList<>();
+
+    public RectangularMap(int width, int height) {
+        this.WIDTH = width;
+        this.HEIGHT = height;
+        this.ORIGIN = new Vector2d(0,0);
+        this.BOUNDARY = new Vector2d(this.WIDTH,this.HEIGHT);
+    }
+
+    @Override
+    public boolean canMoveTo(Vector2d position) {
+        return position.follows(this.ORIGIN) && position.precedes(this.BOUNDARY);
+    }
+
+    @Override
+    public boolean place(Animal animal) {
+        Vector2d pos = animal.getPosition();
+        if (canMoveTo(pos) && !isOccupied(pos)) {
+            this.animals.add(animal);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isOccupied(Vector2d position) {
+        return this.animals.stream().anyMatch(animal -> position.equals(animal.getPosition()));
+    }
+
+    @Override
+    public Object objectAt(Vector2d position) {
+        return this.animals.stream().filter(animal -> position.equals(animal.getPosition())).findFirst().orElse(null);
```

Comment:
taka uwaga na boku do streamów: warto układać je wizualnie trzymając się zasady "nowa kropka od nowej linii", wtedy jest to bardziej czytelne i łatwiejsze w debugowaniu